app/javascript/features/lobby/display/BattleCard.jsx

```diff
@@ -63,6 +64,61 @@ export const BattleCard = ({ id, rulesetId }) => {
   const axisPlayers = addPlaceholders(battle.battlePlayers.filter(p => p.side === 'axis'), size)
   const generatingContent = battle.state === GENERATING ? <Typography>Generating</Typography> : ""
   const ingameContent = battle.state === INGAME ? <Link to={`/api/battles/${id}/battlefiles/zip`} target="_blank" download>Download Battlefile</Link> : ""
+  const balanceColor = (balance) => {
+    switch (balance) {
+      case "Perfect":
+        return "gold";
+      case "Balanced":
+        return "green";
+      case "Allied Stomp":
+        return "blue";
+      case "Axis Stomp":
+        return "red";
+      case "Allied Favoured":
+        return "teal";
+      case "Axis Favoured":
+        return "pink";
+      default:
+        return "white";
+    }}
+  const minDiff = (size) => {
+    if (size == 4)
+      return 50
+    else
+      return 35
+  }
+
+  const balanceState = (balance) => {
+    const min = minDiff(battle.size)
+    const val = Math.abs(balance)
+    if (!balance)
+      return "Pending"
+
+    if (balance < (0-100))
+      return "Allied Stomp"
+    else if (balance > 100)
+      return "Axis Stomp"
+    else if (balance < (0-min))
+      return "Allied Favoured"
+    else if (balance > min)
+      return "Axis Favoured"
+    else if (balance < 10)
+      return "Perfect"
+
+    else
+      return "Balanced"
+  };
+  var optimumBalance = false;
+  var k = 0;
+  // Loop through axis players and check sum of team balance equals match size or twice match size
+  for (let i = 0; i < axisPlayers.length; i++) {
+    k += axisPlayers[i].teamBalance;
+  }
```

Comment:
Could use array.reduce if you wanted to make it pretty

```suggestion
  axisPlayers.reduce((accumulator, currentValue) => accumulator + currentValue.teamBalance, 0)
```